submissions/BlueLamantine/oop/index.js

```diff
@@ -0,0 +1,133 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/BlueLamantine/a-tiny-JS-world
+   Web app: https://bluelamantine.github.io//a-tiny-JS-world/
+   */
+class General {
+  constructor({ name, gender, friends, species }) {
+    this.name = name;
+    this.gender = gender;
+    this.species = species;
+    this.friends = Array.isArray(friends) ? friends : 'no friends';
+  }
+  getGender() {
+    return {
+      male: ['He', 'his'],
+      female: ['She', 'her'],
+    };
+  }
+  getInfo() {
+    return (
+      `This ${this.species} is a ${this.gender}, ` +
+      `${this.getGender()[this.gender][1]} name is ${this.name}. `
+    );
+  }
+  getFriendsList() {
+    return `Friend with: ${this.friends}`;
+  }
+}
+
+class Human extends General {
+  constructor({ name, gender, friends, voise }) {
+    super({ name, gender, friends, species: 'human' });
+    this.voise = voise;
+    this.legs = 2;
+    this.hands = 2;
+  }
+
+  getInfo() {
+    return (
+      super.getInfo() +
+      ` ${super.getGender()[this.gender][0]} has ${this.legs} legs and ${
```

Comment:
By default class inherits parent's properties and methods.
We can refer these by `this.propName` or `this.methodName()` as if they were this class' own props and methods. We just need good reasoning for this, but in general it is legit.

So we can call `.getGender` as `this.getGender()`.

So, why do we call `.getInfo` as `super.getInfo()`? Because it is overloaded and we want call parent's method explicitly.

Take a look at the following fragment:
```
class A {
  constructor() {}
  getNestingLevel() { return 1; }
}

class B {
  constructor() { 
    super();
    this.getNestingLevel(); // 2
    super.getNestingLevel(); // 1
  }
  getNestingLevel() { return 2; }
  someMethod() {}
}

class C extends B {
  contsructor() {
    super();
    this.getNestingLevel(); // 2
    super.getNestingLevel(); // 2
  }
  someMethod() {
    super.someMethod(); // all good
    this.someMethod(); // current method call; leads to an infinite loop
  }
}
```

Reply:
Step by step I feel more confident with oop but still not strong in it